src/snowflake/cli/plugins/git/commands.py

```diff
@@ -0,0 +1,239 @@
+import logging
+from pathlib import Path
+
+import typer
+from click import ClickException
+from snowflake.cli.api.commands.flags import (
+    identifier_argument,
+    like_option,
+    pattern_option,
+)
+from snowflake.cli.api.commands.snow_typer import SnowTyper
+from snowflake.cli.api.console.console import cli_console
+from snowflake.cli.api.constants import ObjectType
+from snowflake.cli.api.output.types import CommandResult, QueryResult
+from snowflake.cli.api.utils.path_utils import is_stage_path
+from snowflake.cli.plugins.git.manager import GitManager
+from snowflake.cli.plugins.object.manager import ObjectManager
+from snowflake.connector import ProgrammingError
+
+app = SnowTyper(
+    name="git",
+    help="Manages git repositories in Snowflake.",
+)
+log = logging.getLogger(__name__)
+
+
+def _repo_path_argument_callback(path):
+    if not is_stage_path(path):
+        raise ClickException(
+            "REPOSITORY_PATH should be a path to git repository stage with scope provided."
+            " For example: @my_repo/branches/main/"
+        )
+    return path
+
+
+RepoNameArgument = identifier_argument(sf_object="git repository", example="my_repo")
+RepoPathArgument = typer.Argument(
+    metavar="REPOSITORY_PATH",
+    help=(
+        "Path to git repository stage with scope provided."
+        " Path to the repository root must end with '/'."
+        " For example: @my_repo/branches/main/"
+    ),
+    callback=_repo_path_argument_callback,
+)
+PatternOption = pattern_option(
+    help_example='`list-files --pattern ".*\.txt"` lists all files with .txt extension',
+)
+
+
+def _object_exists(object_type, identifier):
+    try:
+        ObjectManager().describe(
+            object_type=object_type.value.cli_name,
+            name=identifier,
+        )
+        return True
+    except ProgrammingError:
+        return False
```

Comment:
Consider moving this to SqlManager as this is a generic utility that will be often handful.

Reply:
I moved it to ObjectManager - it strongly depends on "describe" operation, which has common parts with other object commands